app/views/ShareListView/index.js

```diff
@@ -0,0 +1,445 @@
+			this.setState({ searching: false });
+			navigation.setParams({ searching: false });
+			closeSearchHeader();
+			this.internalSetState({ search: [] });
+			Keyboard.dismiss();
+		}
+	}
+
+	renderListHeader = () => <SearchBar onChangeSearchText={this.search} />;
+
+	renderSectionHeader = header => (
+		<View style={styles.headerContainer}>
+			<Text style={styles.headerText}>
+				{I18n.t(header)}
+			</Text>
+		</View>
+	)
+
+	renderItem = ({ item }) => {
+		const { userId, token, baseUrl } = this.props;
+		return (
+			<ShareItem
+				userId={userId}
+				token={token}
+				baseUrl={baseUrl}
+				type={item.t}
+				name={this.getRoomTitle(item)}
+				onPress={() => this.shareMessage(item)}
+			/>
+		);
+	}
+
+	renderSeparator = () => <View style={styles.separator} />;
+
+	renderSection = (data, header) => {
+		if (data && data.length > 0) {
+			return (
+				<View>
+					{this.renderSectionHeader(header)}
+					<View style={styles.bordered}>
+						<FlatList
+							data={data}
+							keyExtractor={keyExtractor}
+							style={styles.flatlist}
+							renderItem={this.renderItem}
+							ItemSeparatorComponent={this.renderSeparator}
+							getItemLayout={getItemLayoutServer}
+							enableEmptySections
+							removeClippedSubviews
+							keyboardShouldPersistTaps='always'
+							initialNumToRender={12}
+							windowSize={20}
+						/>
+					</View>
+				</View>
+			);
+		}
+		return null;
+	}
+
+	renderServerSelector = () => {
+		const { servers } = this.state;
+		const { server } = this.props;
+		const currentServer = servers.find(serverFiltered => serverFiltered.id === server);
+		return currentServer ? (
+			<React.Fragment>
+				{this.renderSectionHeader('Select_Server')}
+				<View style={styles.bordered}>
+					<ServerItem
+						server={server}
+						onPress={() => Navigation.navigate('SelectServerView')}
+						item={currentServer}
+						disclosure
+					/>
+				</View>
+			</React.Fragment>
+		) : null;
+	}
+
+	renderContent = () => {
+		const {
+			discussions, channels, privateGroup, direct, livechat, search
+		} = this.state;
+
+		if (search.length > 0) {
+			return (
+				<FlatList
+					data={search}
+					extraData={search}
+					keyExtractor={keyExtractor}
+					style={styles.flatlist}
+					renderItem={this.renderItem}
+					getItemLayout={getItemLayoutShare}
+					ItemSeparatorComponent={this.renderSeparator}
+					enableEmptySections
+					removeClippedSubviews
+					keyboardShouldPersistTaps='always'
+					initialNumToRender={12}
+					windowSize={20}
+				/>
+			);
+		}
+
+		return (
+			<View style={styles.content}>
+				{this.renderServerSelector()}
+				{this.renderSection(discussions, 'Discussions')}
+				{this.renderSection(channels, 'Channels')}
+				{this.renderSection(direct, 'Direct_Messages')}
+				{this.renderSection(privateGroup, 'Private_Groups')}
+				{this.renderSection(livechat, 'Livechat')}
+			</View>
+		);
+	}
+
+
+	renderScrollView = () => {
```

Comment:
We should check user's settings `group by type`, `group by favorite` and `order`.
During my tests, I find odd to show these groups when I always use without any of them.